compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/BuildProject.java

```diff
@@ -111,4 +114,38 @@ private BuildProject(ProjectEnvironmentBuilder environmentBuilder, Path projectP
         }
         return Optional.empty();
     }
+
+    @Override
+    public DocumentId documentId(Path file) {
+        ProjectPaths.findPackageRoot(file);
+        Path parent = Optional.of(file.toAbsolutePath().getParent()).get();
+        for (ModuleId moduleId : this.currentPackage().moduleIds()) {
+            String moduleDirName;
+            if (moduleId.moduleName().contains(DOT)) {
```

Comment:
It is not correct to get the module name from the `moduleId`. Use `moduleId` to get the ModuleContext.

What's inside moduleId is just for debugging purposes.

Reply:
`ModuleId.moduleName()` is public method which is not correct. Please remove that as well.

Reply:
ModuleContext is not visible to the BuildProject. What should we do in that case?

Reply:
We need to restructure the Java packages in `io.ballerina.projects` to make it work. We can move whatever in the `directory` package to the `io.ballerina.projects` package.

Reply:
Let's go ahead with the current implementation for now. We need to refactor the packages to fix this properly